src/Abstractions/NexusMods.Abstractions.Downloads/ADownloadActivity.cs

```diff
@@ -0,0 +1,101 @@
+using JetBrains.Annotations;
+using NexusMods.Abstractions.Activities;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+
+namespace NexusMods.Abstractions.Downloads;
+
+/// <summary>
+/// Implementation of <see cref="IDownloadActivity"/> for ease of use.
+/// </summary>
+[PublicAPI]
+public abstract class ADownloadActivity : ReactiveObject, IDownloadActivity
+{
+    private readonly IConnection _connection;
+
+    /// <inheritdoc/>
+    public PersistedDownloadStateId PersistedStateId { get; }
+
+    /// <inheritdoc/>
+    public IDownloader Downloader { get; }
+
+    /// <inheritdoc/>
+    public string Title { get; }
+
+    /// <inheritdoc/>
+    public AbsolutePath DownloadPath { get; }
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    protected ADownloadActivity(
```

Comment:
Right, question I forgot to ask, who is supposed to create the `DownloadActivity` (e.g. NxmDownloadActivity())?
- Would it be the initiator of the download like the nxmHandler? 
- Or the Library with something like CreateNxmDL?
- Or the Downloader with something like NXMDownloader.CreateDL()?

Reply:
The NXM protocol handler will create a `NXMDownloadActivity` and enqueue it using `LibraryService.EnqueueDownload`.

Reply:
I guess you would usually leave the creation of things like this to a dedicated service (be it the Downloader or the Library or something else), which would allow more decupling in case in the future you need to introduce further variations (subclasses).